sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -971,14 +971,14 @@ public String getId() {
         PartitionKey partitionKey,
         CosmosQueryRequestOptions options,
         Class<T> classType) {
-
+        final CosmosQueryRequestOptions requestOptions = options == null ? new CosmosQueryRequestOptions() : options;
```

Comment:
Don't we need to set the partition key to the newly created `CosmosRequestOptions` ?

Reply:
I don't think so, looking at the [`readAllDocuments`](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java#L2303) which being called from [here](https://github.com/Azure/azure-sdk-for-java/pull/20744/files#diff-8570ae80eedd164d4433a4d3589e4c4e840a539e5c5190de75f76de06e767a9dR981), it looks like the `partitionKey` is being used separately, not as part of the options like the other service methods (by calling `ModelBridgeInternal.setPartitionKey(options, partitionKey);`  which I think that is not needed here). However, I could be wrong though from other place that I didn't see

Reply:
@omarsmak - right, but I am looking at this public API - readAllItems - which is also setting it in the options here - https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java#L951
Because there are some cases, where we get the partition key from the options in implementation. 
Let's please set it. 

Adding @mbhaskar  @xinlian12  to confirm this behavior.

Reply:
ah true, then I will set it to match the other public APIs

Reply:
@moderakh  - can you please confirm the above behavior ?

Reply:
LGTM. thanks for the bugfix @omarsmak